lib/src/main/kotlin/libTree/trees/AVLTree.kt

```diff
@@ -0,0 +1,287 @@
+package libTree.trees
+
+import libTree.interfaceTree.Tree
+import kotlin.math.max
+
+/**
+ * Implementation of an AVL Tree, a self-balancing binary search tree.
+ *
+ * @param K the type of keys maintained by this tree, must be Comparable
+ * @param V the type of mapped values
+ */
+class AVLTree<K : Comparable<K>, V> private constructor (
+    private var root: AVLNode<K, V>? = null,
+) : Tree<K, V, AVLTree.AVLNode<K, V>> {
+    /**
+     * Node class representing each node in the AVL Tree.
+     *
+     * @param K the type of keys
+     * @param V the type of values
+     * @property left Left child node
+     * @property right Right child node
+     * @property height Height of the node in the tree
+     */
+    class AVLNode<K : Comparable<K>, V> internal constructor (
+        key: K,
+        value: V,
+        override var left: AVLNode<K, V>? = null,
+        override var right: AVLNode<K, V>? = null,
+        override var height: Long = 1
+    ) : BaseNode<K, V, AVLNode<K, V>>(key, value, left, right, height)
+
+    /**
+     * Secondary constructor initializing an empty AVL Tree.
+     */
+    constructor() : this(null)
+
+    /**
+     * Returns the root node of the AVL tree.
+     *
+     * @return The root node
+     */
+    fun getRoot() : AVLNode<K, V>? = root
+
+    /**
+     * Sets the root node of the AVL tree.
+     *
+     * @param node The new root node
+     */
+    fun setRoot(node: AVLNode<K,V>?) {
+        root = node
+    }
+
+    /**
+     * Calculates the height of the AVL Tree.
+     *
+     * @return Height of the tree
+     */
+    override fun height(): Int {
+        return heightOfTree(root)
+    }
```

Comment:
Немного странно, зачем пробегать по всему дереву для вычисления высоты, если для каждого узла есть поле height с готовым значением.
```suggestion
    override fun height(): Int {
    return root?.height?.toInt() ?: 0
}
```

Reply:
Согласен с правкой